filament/backend/src/opengl/platforms/PlatformCocoaTouchGL.mm

```diff
@@ -108,8 +108,9 @@
 }
 
 Platform::SwapChain* PlatformCocoaTouchGL::createSwapChain(uint32_t width, uint32_t height, uint64_t& flags) noexcept {
-    // TODO: implement headless SwapChain
-    return nullptr;
+    CAEAGLLayer* glLayer = [CAEAGLLayer layer];
+    glLayer.frame = CGRectMake(0, 0, width, height);
+    return (__bridge SwapChain*) glLayer;
```

Comment:
Won't this pointer be invalid when `glLayer` goes out of scope?

Reply:
I think, there seems to be no problem.

I tested it with the code below :

```cpp
filament::SwapChain* swap_chain_ =
      engine_->createSwapChain(argear_view_info.width, argear_view_info.height,
                               filament::SwapChain::CONFIG_TRANSPARENT);

std::cout<<"create swapchain : "<< swap_chain_ << std::endl;
```

printed : 
```
create swapchain : 0x1055217a0
```

Reply:
`[CAEAGLLayer layer]` returns an autoreleased object. Since nothing is strongly referencing `glLayer`, it will get released at some point after this method returns. The `SwapChain*` pointer will then be pointing to dead memory, which could lead to a crash.

We need to somehow hold a reference to the layer for the lifetime of the `SwapChain`.

Reply:
Instead of using `__bridge`, use `CFBridgingRetain` to keep the layer alive. Then inside `destroySwapChain`, use `CFBridgingRelease`.

Reply:
Oh, I see.
For some reason, I thought it was okay because the memory was alive even when connected to the c++ code.
I overlooked the instability.

I'll use `CFBridgingRetain `and `CFBridgingRelease`.